Testing/SystemTests/tests/framework/ISIS_PowderOsirisTest.py

```diff
@@ -176,3 +191,97 @@ def _gen_required_files(self):
         input_files = [os.path.join(input_dir, file) for file in required_run_files]
         input_files.append(calibration_map_path)
         return input_files
+
+
+class OSIRISDiffractionFocusingWithSimpleAbsorptionCorrection(systemtesting.MantidSystemTest):
+    existing_config = config["datasearch.directories"]
+
+    def requiredFiles(self):
+        return self._gen_required_files()
+
+    def runTest(self):
+        setup_mantid_paths()
+        sample_details = SampleDetails(radius=1.1, height=8, center=[0, 0, 0], shape="cylinder")
+        sample_details.set_material(chemical_formula="Cr2-Ga-N", number_density=10.0)
+        sample_details.set_container(chemical_formula="Al", number_density=2.7, radius=1.2)
+
+        self.results = run_diffraction_focusing(
+            "120032",
+            "Test_Simple_Absorb",
+            "OSI120032_d_spacing.nxs",
+            absorb_corrections=True,
+            empty_can_subtraction_method="Simple",
+            sample_details=sample_details,
+            simple_events_per_point=100,
+            vanadium_normalization_cutoff=100000000,
+        )
+
+    def validate(self):
+        foccussed_ws = self.results
+        return (foccussed_ws.name(), "OSI120032_d_spacing_simple_corrected.nxs")
+
+    def cleanup(self):
+        try:
+            _try_delete(output_dir)
+        finally:
+            mantid.mtd.clear()
+            config["datasearch.directories"] = self.existing_config
+
+    def _gen_required_files(self):
+        required_run_files = [
+            "OSI82717.nxs",
+            "OSI82718.nxs",  # empty can
+            "OSIRIS00119963.nxs",
+            "OSIRIS00119964.nxs",  # van
+            "OSIRIS00120032.nxs",
+        ]  # sample
+        input_files = [os.path.join(input_dir, file) for file in required_run_files]
+        input_files.append(calibration_map_path)
+        return input_files
+
+
+class OSIRISDiffractionFocusingWithPaalmanPingsAbsorptionCorrection(systemtesting.MantidSystemTest):
+    existing_config = config["datasearch.directories"]
+
+    def requiredFiles(self):
+        return self._gen_required_files()
+
+    def runTest(self):
+        setup_mantid_paths()
+        sample_details = SampleDetails(radius=1.1, height=8, center=[0, 0, 0], shape="cylinder")
+        sample_details.set_material(chemical_formula="Cr2-Ga-N", number_density=10.0)
+        sample_details.set_container(chemical_formula="Al", number_density=2.7, radius=1.2)
+
+        self.results = run_diffraction_focusing(
+            "120032",
+            "Test_PaalmanPings_Absorb",
+            "OSI120032_d_spacing.nxs",
+            absorb_corrections=True,
+            empty_can_subtraction_method="PaalmanPings",
```

Comment:
can we also have a test for the simple option too?

Reply:
it is already added [here](https://github.com/mantidproject/mantid/pull/35430/commits/ead9a22e432a42a7320a4801cc99a313c46aae47#diff-03fc0a417bea1c4853ca17a6a6f57470462db727b3d08497d9762f0258d4c366R196)

Reply:
is there a reason they are not two tests in the same class?

Reply:
I think from a separation of concerns perspective, it is better to implement every test in its own class. Also, it will be clearer to see every test separately in the test output

Reply:
most tests in Manitd use one class with different options. Having a class for each one makes the file very long (repeated boiler plate) and makes it easy to miss things (e.g. me not spotting that you had the test)

Reply:
Right, there is a lot of duplicate code. From my understanding, I guess using one class is the case in C++ but in Python, it is not. Because the test interface is designed to run one test you need to implement `runTest` and `validate`. So it will be strange to run two tests in `runTest` function. To reduce code duplication I can write a common test class for all the tests as in [Pearl Tests](https://github.com/mantidproject/mantid/blob/34311_add_absorbtion_corrections/Testing/SystemTests/tests/framework/ISIS_PowderPearlTest.py). What do you think?

Reply:
Ah ok, its a system test and not a unit test. Sorry I missed that, hence my comments.

A template class would help. I think thats a good idea.

Reply:
No problem, ok, I will do that